src/resource-archive/index.ts

```diff
@@ -26,6 +26,12 @@ class Watcher {
 
   private globalNetworkTimeoutMs;
 
+  /** 
+   Specifies which domains (origins) we should archive resources for (by default we only archive same-origin resources).
+   Useful in situations where the environment running the archived storybook (e.g. in CI) may be restricted to an intranet or other domain restrictions
+  */
+  private allowedExternalDomains: string[];
```

Comment:
Maybe I should be more explicit here that we expect them to supply the full origin (not just the hostname).

Reply:
What do you mean by that? Would those be different?

Reply:
I'm wondering if we should have them specify the entire origin (protocol, hostname, and port) or just the hostname + port. We could tack `https` on the beginning, though if someone were using `localhost` as an external domain for resources that'd break.

But I guess having `localhost` as an external domain is pretty unlikely.

@tevanoff do you think it's easier for the user to specify the entire origin or just the host?

Reply:
I think it's best to just specify the `host`. I don't see any reason why we'd only want to archive assets from an `https` protocol and not from the `http` protocol on the same host.